tiledb/ml/models/base.py

```diff
@@ -9,13 +15,125 @@ class TileDBModel(abc.ABC):
     store machine learning models (Tensorflow, PyTorch, ect) as TileDB arrays.
     """
 
-    def __init__(self, uri: str):
+    def __init__(self, uri: str, ctx=None, namespace=None):
         """
         Base class for saving machine learning models as TileDB arrays
         and loading machine learning models from TileDB arrays.
         :param uri: str
         """
         self.uri = uri
+        self.tiledb_uri = None
+        self.ctx = ctx
+        self.namespace = namespace
+
+        if self.namespace and self.ctx:
+            s3_prefix = self.get_s3_prefix()
+            if s3_prefix is None:
+                raise HTTPError(
+                    403,
+                    "You must set the default s3 prefix path for ML models in {} profile settings".format(
+                        namespace
+                    ),
+                )
+
+            self.uri = "tiledb://{}/{}".format(
+                namespace, os.path.join(s3_prefix, uri)
+            )
+
+            self.tiledb_uri = "tiledb://{}/{}".format(namespace, uri)
+        else:
+            self.uri = uri
+
+    def get_s3_prefix(self):
```

Comment:
We should follow Python 3 standards here and define output type. Moreover, I think that all s3 oriented functionality should go in a different place, e.g, cloud_utils.py or something similar.

Reply:
`cloud_utils.py` was introduced, s3, get model list live there